lib/repo.c

```diff
@@ -77,7 +114,44 @@ repo_get_dict(struct xbps_repo *repo)
 		    archive_error_string(repo->ar));
 		return NULL;
 	}
-	return xbps_archive_get_dictionary(repo->ar, entry);
+	dict = xbps_archive_get_dictionary(repo->ar, entry, &bytes);
+	idxmeta = (repo->idxmeta != NULL) ? repo->idxmeta : dict;
+	if (verify_error != NULL && bytes != NULL) {
+		digest = xbps_buffer_hash_raw(bytes, strlen(bytes));
+		*verify_error = repo_verify_index(repo, idxmeta, digest);
+	}
+	free(digest);
+	free(bytes);
+	return dict;
+}
+
+
+xbps_dictionary_t HIDDEN
+get_safe_idxmeta(xbps_dictionary_t full) {
+	static const char *keys[] = {
+		"public-key",
+		"public-key-size",
+		"signature-by",
+		"signature-type",
```

Comment:
What do these fields represent?

Reply:
They describe which key in `/var/db/xbps/keys/` should be used to verify binary packages.